document/core/exec/instructions.rst

```diff
@@ -464,6 +998,98 @@ Memory Instructions
    \end{array}
 
 
+.. _exec-load-extend:
+.. _exec-load-splat:
+
+:math:`\V128\K{.}\LOAD{M}\K{x}L\_\sx~\memarg` and :math:`\V128\K{.}\LOAD{N}\K{\_splat}~\memarg`
+...............................................................................................
+
+1. Let :math:`F` be the :ref:`current <exec-notation-textual>` :ref:`frame <syntax-frame>`.
+
+2. Assert: due to :ref:`validation <valid-load-extend>`, :math:`F.\AMODULE.\MIMEMS[0]` exists.
+
+3. Let :math:`a` be the :ref:`memory address <syntax-memaddr>` :math:`F.\AMODULE.\MIMEMS[0]`.
+
+4. Assert: due to :ref:`validation <valid-load-extend>`, :math:`S.\SMEMS[a]` exists.
+
+5. Let :math:`\X{mem}` be the :ref:`memory instance <syntax-meminst>` :math:`S.\SMEMS[a]`.
+
+6. Assert: due to :ref:`validation <valid-load-extend>`, a value of :ref:`value type <syntax-valtype>` |I32| is on the top of the stack.
+
+7. Pop the value :math:`\I32.\CONST~i` from the stack.
+
+8. Let :math:`\X{ea}` be the integer :math:`i + \memarg.\OFFSET`.
+
+9. If :math:`N` is not part of the instruction, then:
+
+   a. Let :math:`N` be :math:`M * L`.
+
+10. If :math:`\X{ea} + N/8` is larger than the length of :math:`\X{mem}.\MIDATA`, then:
+
+    a. Trap.
+
+11. Let :math:`b^\ast` be the byte sequence :math:`\X{mem}.\MIDATA[\X{ea} \slice N/8]`.
+
+12. If :math:`\K{splat}` is part of the instruction, then:
+
+    a. Let :math:`n` be the integer for which :math:`\bytes_{\iN}(n) = b^\ast`.
+
+    b. Let :math:`L` be the integer :math:`128 / N`.
+
+    c. Let :math:`c` be the result of computing :math:`\simdto^{-1}_{\iN\K{x}L}(n^L)`.
+
+12. Else:
+
+    a. Let :math:`m_i` be the integer for which :math:`\bytes_{\iM}(m_i) = b^\ast[i*M/8 \slice M/8]`.
```

Comment:
Why can't this just be `b*`?

Reply:
`b*` is the entire 8 bytes that is loaded, we want to extend it lane-by-lane (`m_i`), hence only taking a slice of size `M/8`.

Reply:
Ah, that makes sense. Thanks!